packages/types/src/extrinsic/Doughnut.ts

```diff
@@ -12,32 +12,80 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import {AccountId, Signature, Struct, Text, Tuple, U32, U64, Vector} from '@plugnet/types';
-
-export class Certificate extends Struct {
-    constructor(value?: any) {
-        super(
-            {
-                expires: U64,
-                version: U32,
-                holder: AccountId,
-                notBefore: U64,
-                permissions: Vector.with(Tuple.with([Text, Text])),
-                issuer: AccountId,
-            },
-            value
-        );
+import {u8aToHex} from '@plugnet/util';
+
+/**
+ * An encoded, signed v0 Doughnut certificate
+ *
+ * Doughnut has its own binary codec. We hide this behind the parity codec interface
+ * to support integration as an extrinsic data/field.
+ *
+ **/
+export class Doughnut extends Uint8Array {
```

Comment:
should extends from a Codec type, try `U8a`
then you can remove most of the duplicate codes